librz/core/agraph.c

```diff
@@ -2731,15 +2735,15 @@ static void agraph_set_layout(RzAGraph *g) {
 		if (a->is_dummy) {
 			continue;
 		}
-		const char *k;
-		k = sdb_fmt("agraph.nodes.%s.x", a->title);
-		sdb_num_set(g->db, k, rebase(g, a->x), 0);
-		k = sdb_fmt("agraph.nodes.%s.y", a->title);
-		sdb_num_set(g->db, k, rebase(g, a->y), 0);
-		k = sdb_fmt("agraph.nodes.%s.w", a->title);
-		sdb_num_set(g->db, k, a->w, 0);
-		k = sdb_fmt("agraph.nodes.%s.h", a->title);
-		sdb_num_set(g->db, k, a->h, 0);
+		char buf[1024] = { 0 };
+		rz_strf(buf, "agraph.nodes.%s.x", a->title);
```

Comment:
`agraph.nodes..w` is 15 chars long, if we count also `\0` then we have 16 bytes.
probably `%s` should be limited to `1000` chars.

Reply:
the same for other stuff, but i guess doesn't matter, since `sdb_fmt` was limited to 512 bytes.

Reply:
Just saw that `x->title` is at some point truncated to 255 bytes. So lets limit it to 512?

Reply:
then yes you can move the buffer down.